hudi-common/src/test/java/org/apache/hudi/common/functional/TestHoodieLogFormat.java

```diff
@@ -769,6 +772,36 @@ public void testAppendAndReadOnCorruptedLog() throws IOException, URISyntaxExcep
     reader.close();
   }
 
+  @Test
+  public void testSkipCorruptedCheck() throws Exception {
+    // normal case: if the block is corrupted, we should be able to read back a corrupted block
+    Reader reader1 = createCorruptedFile("test-fileid1");
+    HoodieLogBlock block = reader1.next();
+    assertEquals(HoodieLogBlockType.CORRUPT_BLOCK, block.getBlockType(), "The read block should be a corrupt block");
+    reader1.close();
+
+    // mock case: we want to verify that isBlockCorrupted() check is skipped for transactional write fs
+    // so here created a corrupted block for a transactional write fs(which would never happen) to verify the skip logic
+    // a better way to verify it is to mock the private method isBlockCorrupted(), but it is not available in current Mockito framework
+    // and the attempt to onboard powermock with private method mocking capability has failed
```

Comment:
We can omit the part regarding powermock, and just elaborate that we're doing this b/c we can't mock private method